packages/eas-cli/src/credentials/manager/ManageAndroid.ts

```diff
@@ -61,8 +61,14 @@ export class ManageAndroid {
       user: await ensureLoggedInAsync({ nonInteractive: false }),
       env: buildProfile?.env,
     });
+
+    const getAccountNameForProjectAsync = async (): Promise<string> => {
+      const projectId = await getProjectIdAsync(ctx.exp, { nonInteractive: false });
+      return (await getOwnerAccountForProjectIdAsync(projectId)).name;
+    };
+
     const accountName = ctx.hasProjectContext
-      ? getProjectAccountName(ctx.exp, ctx.user)
+      ? await getAccountNameForProjectAsync()
```

Comment:
If your goal is to remove any reliance on the owner field then getting read of `findAccountByName` calls should be a part of it. That method does not do anything that inherently depends on the owner field in app.json, but as far as I'm aware this is the only case where we know the account name and it's not just a primary account.

In this case, you could do sth like this

```
let account
if (ctx.hasProjectContext) {
      const projectId = await getProjectIdAsync(ctx.exp, { nonInteractive: false });
      account = (await getOwnerAccountForProjectIdAsync(projectId));
} else {
     account = ctx.user.primaryAccount 
}
```

Reply:
Added to task: https://linear.app/expo/issue/ENG-6278#comment-e1966bc6